roxie/udplib/udptrr.cpp

```diff
@@ -466,54 +466,66 @@ class CReceiveManager : implements IReceiveManager, public CInterface
             adjustPriority(1);
         #endif
             UdpRequestToSendMsg msg;
-            unsigned timeout = 5000;  // Is this too long?
+            unsigned timeoutExpires = msTick() + 5000;
             while (running)
             {
                 try
                 {
                     const unsigned l = sizeof(msg);
                     unsigned int res ;
-                    flow_socket->readtms(&msg, l, l, res, timeout);
-                    assert(res==l);
-                    if (udpTraceLevel > 5)
+                    unsigned now = msTick();
+                    if (now >= timeoutExpires)
                     {
-                        StringBuffer ipStr;
-                        DBGLOG("UdpReceiver: received %s msg from node=%s", flowType::name(msg.cmd), msg.sourceNode.getTraceText(ipStr).str());
+                        if (currentRequester)
```

Comment:
I think this now means that if a send takes > udpRequestToSendAckTimeout it will get timedOut.  Is that the intent?

Reply:
The receiver will allow others to send data to it after that time, but the original send's would still be processed if it arrived after that time.

Reply:
ok